### PR TITLE
feature/AT-UN-UI-Course-Validation-GET

### PR DESCRIPTION
**Pull request by Ignacio-jipl97**
...

### CODE REVIEW

File: src/controllers/admin.controller.js [hunk 1]
```diff
@@ -5,26 +5,30 @@ const universityServiceAPI = require("../services/at-university-api.service");
...
 };
 
 // AT-UNIVERSITY - Admin - Render Course List
 adminCtrl.renderCourseList = async (req, res) => {
     try {
         const responseCourses = await universityServiceAPI.getAllCourses();
         console.log("---> adminCtrl.renderCourseList.getAllCourses");
-        //console.log(responseCourses.data);
         const courses = responseCourses.data;
-        res.render("admin/course/index", { courses });
```
Comment (at-internship) [inline src/controllers/admin.controller.js]: Why did you delete it?

File: src/controllers/admin.controller.js [hunk 2]
```diff
@@ -5,26 +5,30 @@ const universityServiceAPI = require("../services/at-university-api.service");
...
+        res.render("admin/course", { courses });
     } catch (err) {
-        console.error(err.message);
-        res.render("admin/course/index");
+        //if you need to check more details of the error in console replace ".message" by ".response"
+        console.log(err.message);
+        if (err.response && err.response.data) {
+            let errorMsgs = err.response.data.error;
+            req.flash("error_msg", errorMsgs);
+            res.redirect("/admin");
```
Comment (at-internship) [inline src/controllers/admin.controller.js]: is that correct?

File: src/controllers/admin.controller.js [hunk 3]
```diff
@@ -45,32 +49,36 @@ adminCtrl.addCourse = async (req, res) => {
...
+        let responseCourses = await universityServiceAPI.getAllCourses();
+        courseDetails = responseCourses.data.filter(function (c) { return c._id == courseId; });
+        console.log("One course found", courseDetails[0]);
+        res.render("admin/course/edit-course", courseDetails[0]);
+    } catch (err) {
+        console.log(err.response);
+        if (err.response && err.response.data) {
+            let errorMsg = err.response.data.error;
+            req.flash("error_msg", errorMsg, " Try again later");
+            res.redirect("/admin");
```
Comment (at-internship) [inline src/controllers/admin.controller.js]: is that correct?

File: src/services/at-university-api.service.js [hunk 4]
```diff
@@ -18,8 +18,6 @@ courseService.getAllCourses = () => {
     headers: {
       "content-type": "application/json",
     },
-  }).catch(function (error) {
```
Comment (at-internship) [inline src/services/at-university-api.service.js:21]: Excellent

**Review by at-internship — CHANGES_REQUESTED** [review]

File: src/controllers/admin.controller.js (hunk 2 above)
Comment (raaul-at) [reply src/controllers/admin.controller.js]: Yes :smile:

File: src/controllers/admin.controller.js (hunk 3 above)
Comment (raaul-at) [reply src/controllers/admin.controller.js]: Yes :smile:

File: src/controllers/admin.controller.js (hunk 1 above)
Comment (Ignacio-jipl97) [reply src/controllers/admin.controller.js]: "/ index" is not being executed and with that line or without that line it continues executing without "/ courses"

File: src/controllers/admin.controller.js (hunk 2 above)
Comment (Ignacio-jipl97) [reply src/controllers/admin.controller.js]: Yes is correct

File: src/controllers/admin.controller.js (hunk 3 above)
Comment (Ignacio-jipl97) [reply src/controllers/admin.controller.js]: Yes 😄

File: src/controllers/admin.controller.js (hunk 2 above)
Comment (at-internship) [reply src/controllers/admin.controller.js]: I think no, should be `res.redirect("/admin/course");`

File: src/controllers/admin.controller.js (hunk 3 above)
Comment (at-internship) [reply src/controllers/admin.controller.js]: I think no, should be `res.redirect("/admin/course");`

File: src/controllers/admin.controller.js (hunk 2 above)
Comment (Ignacio-jipl97) [reply src/controllers/admin.controller.js]: ready 😄

File: src/controllers/admin.controller.js (hunk 3 above)
Comment (Ignacio-jipl97) [reply src/controllers/admin.controller.js]: ready 😄

**Review by at-internship — APPROVED** [review]
Approved

**Review by raaul-at — APPROVED** [review]
